### DISCUSSION TITLE
Unable to access FreshRSS feeds via Android

### DISCUSSION THREAD

**Discussion by Alzarath** [FreshRSS/FreshRSS#5533] [Q&A]
I've tried all three RSS readers on Android from F-Droid. FreshRSS, EasyRSS, and Readrops. I've been able to log into each. The FreshRSS app and EasyRSS display all of the categories. None of them succeed at fetching any feeds. Accessing through the web browser on the appropriate and other devices works fine otherwise.

The FreshRSS app displays the error `FreshRSS failed to retrieve content from your...`
The EasyRSS app displays the error `Sync failed: Invalid HTTP status 400: http://IP_HERE`
The Readrops app processes for a while then displays `timeout`.

Running the server on:

```
Raspberry Pi 4B
Manjaro ARM 23.02
lighttpd 1.4.71-1
php 8.2.7-1
php-sqlite 8.2.7-1
sqlite 3.42.0-1
```

There are no errors I see as relevant in `/var/lighttpd/error.log`. FreshRSS's Logs page only has errors related to authentication from times I'd entered incorrect information.

What can I do to debug from here?

**Comment by Alwaysin** [comment]
What do you get when going to `http[s]://<hostname>/api/`? EDIT: Depending on the results, this doc might come in handy: https://freshrss.github.io/FreshRSS/en/users/06_Mobile_access.html

Do you have logs in data/users/_/log_api.txt ?

Is FreshRSS up-to-date - what version do you have?

Maybe also try with Apache or NGINX to rule out any misconfiguration in lighttpd (see ex. confs here: https://freshrss.github.io/FreshRSS/en/admins/10_ServerConfig.html)

**Comment by Alzarath** [comment]
> What do you get when going to `http[s]://<hostname>/api/`? EDIT: Depending on the results, this doc might come in handy: https://freshrss.github.io/FreshRSS/en/users/06_Mobile_access.html

Both give me a PASS.

> Do you have logs in data/users/_/log_api.txt ?

The only two types of messages in there are `GReaderAPI::unauthorized` from when I was changing the password and needed to re-login, and the occasional `GReaderAPI::checkCompatibility`

> Is FreshRSS up-to-date - what version do you have?

The latest release, 1.21.0, yes.

> Maybe also try with Apache or NGINX to rule out any misconfiguration in lighttpd (see ex. confs here: https://freshrss.github.io/FreshRSS/en/admins/10_ServerConfig.html)

An apache server gives me similar results. I did have to make some changes to the example configs to match the system, but with and without the configuration I had the same issue.

An oddity, the FreshRSS app was at some point able to retrieve feeds from the time I posted the original thread while I wasn't paying attention. I have been unable to reproduce this success since, however.

**Comment by Alkarex** [comment]
Hello,
If you have tried multiple clients and checked the logs without identifying the problem, I suggest you try by hand using curl. See some examples of commands:
https://freshrss.github.io/FreshRSS/en/developers/06_GoogleReader_API.html#google-reader-compatible-api

Note that it looks like you are using an IP address instead, which might not be accessible from your mobile phone. The best is to use a public IP with DNS and HTTPS.

**Comment by Alzarath** [comment]
> Note that it looks like you are using an IP address instead, which might not be accessible from your mobile phone. The best is to use a public IP with DNS and HTTPS.

That I could access the hosted FreshRSS page from the browser via IP from my phone does seem to indicate that's not the issue. It was just accessing it via the API.

> I suggest you try by hand using curl. See some examples of commands: https://freshrss.github.io/FreshRSS/en/developers/06_GoogleReader_API.html#google-reader-compatible-api

I did try all the example commands from my PC and the output looked correct.

(Un?)fortunately I've been able to access the instance through Readrops the last couple of days without issue. It's possible I just needed to reboot the device, but that does feel unlikely to me. EasyRSS and FreshRSS still have the same errors, however. I've since moved to FeedMe which also functions.

If there's anything you want from me to try to help debug, I'm willing. Otherwise I'll probably continue using FeedMe going forward.
